crates/ty_python_semantic/src/types.rs

```diff
@@ -5155,6 +5224,24 @@ impl<'db> Type<'db> {
                     .into()
                 }
 
+                Some(KnownClass::NewType) => {
+                    Binding::single(
+                        self,
+                        Signature::new(
+                            Parameters::new([
+                                Parameter::positional_or_keyword(Name::new_static("name"))
+                                    .with_annotated_type(Type::LiteralString),
+                                Parameter::positional_or_keyword(Name::new_static("tp")),
+                            ]),
+                            // The actual return type is a KnownInstanceType::NewType(_), but each
+                            // callsite gets a unique payload, including a `Definition` that points
+                            // to that callsite. See `struct NewType` and `infer_newtype_expression`.
+                            None,
+                        ),
+                    )
+                    .into()
+                }
```

Comment:
one issue here is that on Python 3.9, `NewType` is a function rather than a class. I don't know if this really matters this much TBH, since Python 3.9 is now end-of-life, and supporting both the pre-3.9 version of `NewType` (where it was a function) and the 3.10-version of `NewType` (where it's a class) is going to lead to complications for us.

At the very least, it might be good to add an mdtest with `python-version = "3.9"` so that we document that we don't support `NewType` on Python <=3.9 currently. But I would certainly be inclined to leave <=3.9 support out of the first PR adding support for `NewType`.

Reply:
What's your recommendation for this diagnostic? It doesn't look like I have an `InferContext` at this point to directly emit something. Is there a type that I can return that indirectly forces a diagnostic?